lib/hooks.dart

```diff
@@ -101,3 +101,141 @@ StateHook<T> useState<T>(T initialValue) => StateHook(initialValue);
 ///
 /// Learn more: <https://reactjs.org/docs/hooks-reference.html#lazy-initial-state>.
 StateHook<T> useStateLazy<T>(T init()) => StateHook.lazy(init);
+
+/// The return value of [useReducer].
+///
+/// The current state is available via [state] and action dispatcher is available via [dispatch].
+///
+/// > __Note:__ there are two [rules for using Hooks](https://reactjs.org/docs/hooks-rules.html):
+/// >
+/// > * Only call Hooks at the top level.
+/// > * Only call Hooks from inside a [DartFunctionComponent].
+///
+/// Learn more: <https://reactjs.org/docs/hooks-reference.html#usereducer>.
+class ReducerHook {
+  /// The first item of the pair returned by [React.userReducer].
+  Map _state;
```

Comment:
State/actions don't have to be maps, do they? If not, this class should use generics:
```dart
class ReducerHook<TState, TAActions> {
  TState _state;
  void Function(TActions ) _dispatch;
}
```
as should the useReucer* functions